File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/google/admob/AdsFragment.java
package com.jdroid.android.sample.ui.google.admob;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.view.View;
import android.view.View.OnClickListener;

import com.jdroid.android.activity.ActivityLauncher;
import com.jdroid.android.firebase.admob.interstitial.InterstitialAction;
import com.jdroid.android.fragment.AbstractFragment;
import com.jdroid.android.sample.R;
import com.jdroid.android.sample.firebase.remoteconfig.AndroidRemoteConfigParameter;
import com.jdroid.android.utils.ToastUtils;
import com.jdroid.java.remoteconfig.RemoteConfigParameter;

public class AdsFragment extends AbstractFragment {

	@Override
	public Integer getContentFragmentLayout() {
		return R.layout.ads_fragment;
	}

	@Override
	public void onViewCreated(View view, Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);

		findView(R.id.fragmentBanner).setOnClickListener(new OnClickListener() {

			@Override
			public void onClick(View v) {
				ActivityLauncher.startActivity(getActivity(), FragmentBannerActivity.class);
			}
		});

		findView(R.id.adReycler).setOnClickListener(new OnClickListener() {

			@Override
			public void onClick(View v) {
				ActivityLauncher.startActivity(getActivity(), AdRecyclerActivity.class);
			}
		});

		findView(R.id.activityBanner).setOnClickListener(new OnClickListener() {

			@Override
			public void onClick(View v) {
				ActivityLauncher.startActivity(getActivity(), ActivityBannerActivity.class);
			}
		});

		findView(R.id.displayInterstitial).setOnClickListener(new OnClickListener() {

			@Override
			public void onClick(View v) {
				new InterstitialAction() {
					@Override
					protected void onAction() {
						ToastUtils.showToast(R.string.interstitialAction);
					}

					@NonNull
					@Override
					protected RemoteConfigParameter getEnabledRemoteConfigParameter() {
						return AndroidRemoteConfigParameter.INTERSTITIAL_ENABLED;
					}
				}.start(getActivityIf());
			}
		});

		findView(R.id.houseAds).setOnClickListener(new OnClickListener() {

			@Override
			public void onClick(View v) {
				ActivityLauncher.startActivity(getActivity(), HouseAdsActivity.class);
			}
		});
	}
}

File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/google/admob/AdsFragment.kt
package com.jdroid.android.sample.ui.google.admob

import android.os.Bundle
import android.view.View
import com.jdroid.android.activity.ActivityLauncher
import com.jdroid.android.firebase.admob.interstitial.InterstitialAction
import com.jdroid.android.fragment.AbstractFragment
import com.jdroid.android.sample.R
import com.jdroid.android.sample.firebase.remoteconfig.AndroidRemoteConfigParameter
import com.jdroid.android.utils.ToastUtils
import com.jdroid.java.remoteconfig.RemoteConfigParameter

class AdsFragment : AbstractFragment() {

    override fun getContentFragmentLayout(): Int? {
        return R.layout.ads_fragment
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        findView<View>(R.id.fragmentBanner).setOnClickListener { ActivityLauncher.startActivity(activity, FragmentBannerActivity::class.java) }

        findView<View>(R.id.adReycler).setOnClickListener { ActivityLauncher.startActivity(activity, AdRecyclerActivity::class.java) }

        findView<View>(R.id.activityBanner).setOnClickListener { ActivityLauncher.startActivity(activity, ActivityBannerActivity::class.java) }

        findView<View>(R.id.displayInterstitial).setOnClickListener {
            object : InterstitialAction() {
                override fun onAction() {
                    ToastUtils.showToast(R.string.interstitialAction)
                }

                override fun getEnabledRemoteConfigParameter(): RemoteConfigParameter {
                    return AndroidRemoteConfigParameter.INTERSTITIAL_ENABLED
                }
            }.start(getActivityIf()!!)
        }

        findView<View>(R.id.houseAds).setOnClickListener { ActivityLauncher.startActivity(activity, HouseAdsActivity::class.java) }
    }
}
